### ISSUE TITLE
Filter segments on server initial state only looking in the first facet group

### ISSUE THREAD

**Issue by joaopedrofump** [Farfetch/blackout#906]
## Expected behavior

When there are more than one facet group, when building the filter segments, it should look on all facet groups, not only in the first one. Example: We have 3 categories: Dresses, jumpsuits and knitwear. Each one of these categories has a facetgroup for the sizes, being dresses -> [xs], jumpsuits-> [xs,s,m,l,xl], knitwear -> [s]. 

When building the filter segments only the first array is being used causing for example the s or m size not be found. The code responsible for this is this:

file: `blackout-redux > entities > schemas > productsLists.js`

```if (filterSegment.type === ATTRIBUTES_TYPE) {
        let facetIndex;
        const facetGroup = filteredFacetGroups.find(facet => facet.values[0]?.find(({
          value
        }, i) => {
          if (value === filterSegment.value) {
            facetIndex = i;
            return true;
          }
          return false;
        }));
        const facet = facetIndex ? facetGroup?.values[0]?.[facetIndex] : undefined;
        return {
          ...filterSegment,
          description: filterSegment.description || facet?.description,
          facetId: facet ? getId(facet, facetGroup) : undefined
        };
      }
      const facet = filteredFacetGroups[0]?.values[0]?.find(({
        value
      }) => value === filterSegment.value);
      return {
        ...filterSegment,
        description: filterSegment.description || facet?.description,
        facetId: facet ? getId(facet, filteredFacetGroups?.[0]) : undefined
      };

```
## Actual behavior

To build the filter segments, all the facetgroups should be considered.

There is also a bug here :

`const facet = facetIndex ? facetGroup?.values[0]?.[facetIndex] : undefined;`

If facetIndex is 0 (which means the first position) facet will come as undefined as 0 is a falsy value. The condition should be  `typeof faceIndext !== 'undefined'`

## Steps to reproduce

1. Access [here](https://www-napsandbox2-ft06.blackandwhite-ff.com/en-pt/shopping/clothing?contextfilters=categories:182532&pageindex=1)
2. On the clothing sizes select xs
3. On the clothing sizes select s
4. Check that neither the checkboxes nor the pills are correctly working

**Comment by boliveira** [comment]
Hello @joaopedrofump,
Thank for opening this issue. I acknowledge the bug and we will work on fix for it. When we have the release with the fix available, we will let you know.

**Comment by boliveira** [comment]
Hi @joaopedrofump,

We have opened this PR which will address this issue: https://github.com/Farfetch/blackout/pull/913

**Comment by boliveira** [comment]
Hi @joaopedrofump,

The fix for this issue was released last week and is available on the `@farfetch/blackout-redux@1.15.0` package.
Please, reopen this issue if you find any problems.

Thank you.
